### ISSUE TITLE
More fields for ban endpoints

### ISSUE THREAD

**Issue by deadsurgeon42** [NyxStudios/TShock#1413]
Would allow making a ban list with ban date and banning staff without querying database.

Ready for PR:
https://github.com/deadsurgeon42/TShock/blob/787e5e0108e6c1e2d82a9e805e6ad51b0128c0da/TShockAPI/Rest/RestManager.cs#L729

**Comment by hakusaro** [comment]
Really hard to read a diff without a diff...

**Comment by deadsurgeon42** [comment]
Pls I dunno how to do that but is this it:
https://github.com/NyxStudios/TShock/compare/general-devel...deadsurgeon42:general-devel

**Comment by hakusaro** [comment]
I just spent the last 5 minutes trying to figure out what you were trying to accomplish and how that correlated with "more fields for ban endpoints." In the future, please type a human readable description of what and why you changed something so that I don't have to turn on my internal C# parser at the early hour of the morning that I'm working at.

That being said, this seems reasonable to me -- though changing the output of an existing route will break existing implementations. I'd say that it needs to have a version upgrade for it to add supplementary info (cc @WhiteXZ?)

Also, triaging this feature request appropriately.

**Comment by hakusaro** [comment]
FFN: @deadsurgeon42 wants to add banner & ban time to the list route.

**Comment by WhiteXZ** [comment]
@hakusaro I believe in private discussion we determined that *adding* output to an existing route was ok, but *removing* or *modifying* existing output would require an upgrade.
Reasoning being that adding shouldn't break any existing implementations, because all of the previously expected data still exists

**Comment by hakusaro** [comment]
@WhiteXZ while that's totally true, I don't know exactly what's being returned in this JSON response. Since it's a list of objects, it might change the structure unexpectedly if people are deserializing the object.

Though I'm still all for adding those fields. We should really publish docs on how to and how to not read the REST data so that we can be 100% confident in these decisions.

tl;dr: If you think it's okay we can totally do it.

**Comment by WhiteXZ** [comment]
Unless someone has a janky manual json string parser set up for reading out REST API, adding new lines to the response *shouldn't* cause issues, as JSON parsers simply ignore fields they're not told to read

**Comment by ProfessorXZ** [comment]
This seems to be handled in [@deadsurgeon42 's Pull Request](https://github.com/NyxStudios/TShock/pull/1414). Can be reopened if needed.
